contracts/WrapperRegistry.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.4;
+
+import {IWrapperRegistry} from "./interfaces/IWrapperRegistry.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+/**
+ * @title The Wrapper Registry
+ *
+ * @dev Registers bijective pairs of tokens and button/unbutton wrappers
+ *
+ */
+contract WrapperRegistry is IWrapperRegistry, Ownable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /**
+     * @dev Mapping of underlying tokens to wrapper tokens
+     */
+    mapping(address => address) private _wrapperMapping;
```

Comment:
You can make the `_underlyingToWrapperMapping` an `EnumerableMap` instead of the regular mapping and get rid of the `_underlyingTokens` address set.

https://docs.openzeppelin.com/contracts/3.x/api/utils#EnumerableMap

That should still accomplish what you want ie):
* O(1) add, remove and existence checks for underlying tokens
* O(n) enumeration

Reply:
@aalavandhan, I believe EnumerableMap currently only has support for `UintToAddressMap`. That's the reason why I use both `EnumerableSet` and `mapping`  to achieve the same goal.

Reply:
Makes sense 👍